src/variables/components/VariableDropdownReactSelect.tsx

```diff
@@ -0,0 +1,369 @@
+// Libraries
+import React, {PureComponent} from 'react'
+import {connect, ConnectedProps} from 'react-redux'
+import Select from 'react-select'
+
+// Components
+import {MenuStatus} from '@influxdata/clockface'
+
+// Actions
+import {selectValue} from 'src/variables/actions/thunks'
+
+// Utils
+import {getVariable, normalizeValues} from 'src/variables/selectors'
+
+// Types
+import {AppState, RemoteDataState} from 'src/types'
+
+interface OwnProps {
+  variableID: string
+  testID?: string
+  onSelect?: () => void
+}
+
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = OwnProps & ReduxProps
+
+// actualVal keeps track of the actual, selected variable
+// it allows the component to return to that if something was typed that is not in the selected list
+
+// the typed val is what the user types in
+interface MyState {
+  actualVal: string
+  selectIndex: number
+  shownValues: string[]
+  selectHappened: boolean
+  menuOpen: MenuStatus
+  loaded: boolean
+}
+
+class TypeAheadVariableDropdown extends PureComponent<Props, MyState> {
+  constructor(props) {
+    super(props)
+
+    const defaultState = {
+      actualVal: '',
+      selectIndex: -1,
+      shownValues: props.values,
+      selectHappened: false,
+      menuOpen: null,
```

Comment:
null means that the menu status is not changed; just leave it alone.

Reply:
actually...that code is moot.  i lost sight of why i was making this change...which is to have the component manage the state instead of the wrapper manage the state.  and a lot of the extra state that you mentioned can be discarded.  thanks for that insightful comment.  am removing *at least* the selectIndex and menuStatus/menuOpen right now.